docs/050-breaking-changes.rst

```diff
@@ -0,0 +1,379 @@
+  where ``memory`` is the data location and might be replaced by ``storage`` or
+  ``calldata`` accordingly.  Note that ``external`` functions require
+  parameters with a data location of ``calldata``.
+
+* Variables of contract type do not contain ``address`` members anymore in
+  order to separate the namespaces.  Therefore, it is now necessary to
+  explicitly convert values of contract type to addresses before using an
+  ``address`` member.  Example: if ``c`` is a contract, change
+  ``c.transfer(...)`` to ``address(c).transfer(...)``.
+
+* The ``address`` type  was split into ``address`` and ``address payable``,
+  where only ``address payable`` provides the ``transfer`` function.  An
+  ``address payable`` can be directly converted to an ``address``, but the
+  other way around is not allowed. Converting ``address`` to ``address
+  payable`` is possible via conversion through ``uint160``.
+
+* Conversions between ``bytesX`` and ``uintY`` of different size are now
+  disallowed due to ``bytesX`` padding on the right and ``uintY`` padding on
+  the left which may cause unexpected conversion results.  The size must now be
+  adjusted within the type before the conversion.  For example, you can convert
+  a ``bytes4`` (4 bytes) to a ``uint64`` (8 bytes) by first converting the
+  ``bytes4`` variable to ``bytes8`` and then to ``uint64``.
+
+* Using ``msg.value`` in ``non-payable`` functions (or introducing it via a
+  modifier) is disallowed as a security feature. Turn the function into
+  ``payable`` or create a new internal function for the program logic that
+  uses ``msg.value``.
+
+* For clarity reasons, the command line interface now requires ``-`` if the
+  standard input is used as source.
+
+Deprecated Elements
+===================
+
+This section lists changes that deprecate prior features or syntax.  Note that
+many of these changes were already enabled in the experimental mode
+``v0.5.0``.
+
+Command Line and JSON Interfaces
+--------------------------------
+
+* The command line option ``--formal`` (used to generate Why3 output for
+  further formal verification) was deprecated and is now removed.  A new
+  formal verification module, the SMTChecker, is enabled via ``pragma
+  experimental SMTChecker;``.
+
+* The command line option ``--julia`` was renamed to ``--yul`` due to the
+  renaming of the intermediate language ``Julia`` to ``Yul``.
+
+* The ``--clone-bin`` and ``--combined-json clone-bin`` command line options
+  were removed.
+
+* Remappings with empty prefix are disallowed.
+
+* The Json AST fields ``constant`` and ``payable`` were removed. The
+  information is now present in the ``stateMutability`` field.
+
+Constructors
+------------
+
+* Calling base constructors without parentheses is now disallowed.
+
+* Specifying base constructor arguments multiple times in the same inheritance
+  hierarchy is now disallowed.
+
+* Calling a constructor with wrong argument count is now disallowed.
```

Comment:
This is a little misleading. I think the actual change was that if parentheses are given, the arguments have to be provided, too. If you only want to specify an inheritance relation without giving arguments, do not provide parentheses at all.